src/runtime/schema.ts

```diff
@@ -211,11 +225,15 @@ export class Schema {
       if (fields[name] == undefined) {
         return AtLeastAsSpecific.NO;
       }
-      if (type.kind && type.kind === 'schema-nested') {
-        if (!(fields[name].kind && fields[name].kind === 'schema-nested')) {
+      if (type.kind && ['schema-nested', 'schema-reference', 'schema-collection'].includes(type.kind)) {
+        if (!(fields[name].kind && fields[name].kind === type.kind)) {
           return AtLeastAsSpecific.NO;
         }
-        const subResult = fields[name].schema.model.entitySchema.isEquivalentOrMoreSpecific(type.schema.model.entitySchema);
+        const subResult = type.kind === 'schema-collection'
+          ? fields[name].schema.schema.model.entitySchema.isEquivalentOrMoreSpecific(
```

Comment:
Might be worth a comment here, but maybe I just haven't thought about it enough?

Reply:
I rearranged the code a bit, i think it is much more readable now.